server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1243,6 +1247,71 @@ private void markSearcherAccessed() {
         }
     }
 
+    static Engine.Searcher wrapSearcher(Engine.Searcher engineSearcher, IndexReaderWrapper readerWrapper) throws IOException {
```

Comment:
now that we have this can we remove all the cruft from `IndexReaderWrapper`? it seems duplicated? I also wonder if we now can make IndexReaderWrapper a simple `Function<DirectoryReader, DirectoryReader>` and remove it altogether?

Reply:
This is a leftover, thanks. I intended to remove the `IndexReaderWrapper` in a follow up but the change is not so big so I pushed it [here](https://github.com/elastic/elasticsearch/pull/43645/commits/f0fe58924261378c0caffa9127f59507227601ea)